### PR TITLE
Fix Python 3.9+ NamedTuple issues

### LINKED ISSUE #460: Pickling error for `typing.NamedTuple` on Python 3.9
The current `cloudpickle` `main` branch fails to roundtrip a [`typing.NamedTuple`](https://docs.python.org/3/library/typing.html#typing.NamedTuple) object when run on Python 3.9. Here's an example snippet 

```python
import cloudpickle
from typing import NamedTuple

class Record(NamedTuple):
    value: int

result = cloudpickle.dumps(Record(10))
cloudpickle.loads(result)
```

which fails with the following traceback

```python
Traceback (most recent call last):
  File "/Users/james/projects/cloudpipe/cloudpickle/test.py", line 8, in <module>
    cloudpickle.loads(result)
  File "/Users/james/projects/cloudpipe/cloudpickle/cloudpickle/cloudpickle.py", line 823, in _make_skeleton_class
    skeleton_class = types.new_class(
  File "/Users/james/mambaforge/envs/cloudpickle/lib/python3.9/types.py", line 77, in new_class
    return meta(name, resolved_bases, ns, **kwds)
  File "/Users/james/mambaforge/envs/cloudpickle/lib/python3.9/typing.py", line 1852, in __new__
    module=ns['__module__'])
KeyError: '__module__'
```

Interestingly things seem to work fine on Python 3.7 and 3.8. Additionally, I should note that the above snippet also works on Python 3.9 when using the built-in `pickle` module instead of `cloudpickle`. 

This was originally surfaced over in https://github.com/dask/distributed/issues/5602

### PR DESCRIPTION
**Pull request by RyanClark2k**
Reopening #492 here. Fixes #460.

### CODE REVIEW

**Comment by RyanClark2k** [comment]
@ogrisel I closed #492 and reopened here because there was a mixup with my account and another user's (bit of a long story there). I've added the `__slots__` attribute check to `test_instance_with_slots`, and it passed in my local tests. Think we need you or another maintainer to approve the CI run for this new PR though. Cheers!

**Review by ogrisel — APPROVED** [review]
Thanks for the updated PR, LGTM. Could just please document the change in the changelog targeting the next release?

**Comment by RyanClark2k** [comment]
@ogrisel I've updated the changelog and committed the test simplification. Do you know when we can expect the next release? This is the last blocker for my org supporting Python 3.9.

File: CHANGES.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 2.3.0 (in development)
 ======================
```
Comment (ogrisel) [inline CHANGES.md:2]:
```suggestion
2.2.1
=====
```

**Review by ogrisel — APPROVED** [review]
Let's do a bugfix release today with that single fix then.

**Comment by ogrisel** [comment]
cloudpickle 2.2.1 is online on pypi.org and conda-forge should be updated in the coming hours.

**Comment by RyanClark2k** [comment]
@ogrisel much appreciated, I see the new version on Conda Forge. Thanks for your help here!

**Pull request merged**